src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1032,7 +1032,7 @@ public bool TakeStartLine(ReadableBuffer buffer, out ReadCursor consumed, out Re
             var result = _parser.ParseRequestLine(this, buffer, out consumed, out examined);
             if (!result && overLength)
             {
-                RejectRequest(RequestRejectionReason.RequestLineTooLong);
+                RequestRejectionUtilities.RejectRequest(RequestRejectionReason.RequestLineTooLong);
```

Comment:
(the Throwing function will be jitted at the point it is encountered in the calling function; as the jit needs to see what it does; the Exception getter function will only be jitted when the Throwing function itself is called i.e. an exception is being thrown; as it doesn't need to know what it does till then; as its determined its no returning, non-inlinabe function)